### PR TITLE
Add Road Network Disbursement algorithm to geospatial feature

### PR DESCRIPTION
**Pull request by Charl1996**
Same as https://github.com/dimagi/commcare-hq/pull/34114, but with the feature put behind the FF again.

## Product Description
[Demo video](https://vimeo.com/905617454/0f7b334709?share=copy) (note that the FF was removed)

The disbursement algorithm, "Road Network Algorithm",  makes use of the Mapbox Matrix API. This PR adds the ability for a project to add their own mapbox token on the Geospatial Settings page. 

image

A the new disbursement algorithm will appear on the Geospatial Settings page. Once the algorithm is selected, two new HTML elements will appear on the screen:
1) Text (password) input prompting the user for their mapbox token.
2) A button ("Test API Key") which the user can click to make a sample request to Mapbox in order to verify that the token works.

Note that the user don't have to click the "Test API Key" button in order to save the token; the button exists purely to test the token. 

## Technical Summary
[Ticket](https://dimagi-dev.atlassian.net/browse/SC-3360)

Noteworthy items:
1) The api token is stored in postgres as an encrypted string, but returned to the front end in plaintext (although rendered as a password) when loading the Settings page.
2) The "Test API Key" button sample API call happens purely in the browser; I decided we don't have to have a dedicated endpoint just for the server to make an API call to Mapbox if we could simply use ajax.  

## Feature Flag
New FF, `SUPPORT_ROAD_NETWORK_DISBURSEMENT_ALGORITHM`

### Safety story
Tested this feature on my local machine.

### Automated test coverage
- Tests for view
- Tests for model

### QA Plan
QA planned

### Migrations
- [x] The migrations in this code can be safely applied first independently of the code

### Rollback instructions
- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

**Review by zandre-eng — APPROVED** [review]

File: corehq/apps/geospatial/forms.py [hunk 1]
```diff
@@ -164,4 +208,14 @@ def clean(self):
             if not cleaned_data['target_group_count']:
                 raise ValidationError(_("Value for target group count required"))
 
+        algorithm = cleaned_data.get('selected_disbursement_algorithm')
```
Comment (mkangia) [inline corehq/apps/geospatial/forms.py:211]: nit: should we also add a check here to ensure the domain has the feature flag enabled?
I understand its not that critical so even if one bypasses the check on the form and save the configuration with this new algorithm it is not critical but good to do add such checks as a practice.

**Review by mkangia — APPROVED** [review]
Thanks for the attempts on this one @Charl1996. Good that we could think this through well before releasing

File: corehq/apps/geospatial/forms.py (hunk 1 above)
Comment (mkangia) [reply corehq/apps/geospatial/forms.py:211]: >  the form validation fails if the flag is not enabled, because the road_network_algorithm is not added as a valid option for the selected_disbursement_algorithm input.

This is good enough for me. Thanks.
